internal/tui/models/http_request.go

```diff
@@ -0,0 +1,97 @@
+package models
+
+import (
+	"time"
+)
+
+// HTTPRequest represents a captured HTTP request
+type HTTPRequest struct {
+	ID          string            `json:"id"`
+	Method      string            `json:"method"`
+	URL         string            `json:"url"`
+	Path        string            `json:"path"`
+	Headers     map[string]string `json:"headers"`
+	Body        string            `json:"body"`
+	Timestamp   time.Time         `json:"timestamp"`
+	Duration    time.Duration     `json:"duration"`
+	StatusCode  int               `json:"status_code"`
+	Response    *HTTPResponse     `json:"response,omitempty"`
+	ClientIP    string            `json:"client_ip"`
+	UserAgent   string            `json:"user_agent"`
+}
+
+// HTTPResponse represents the response to an HTTP request
+type HTTPResponse struct {
+	StatusCode int               `json:"status_code"`
+	Headers    map[string]string `json:"headers"`
+	Body       string            `json:"body"`
+	Size       int64             `json:"size"`
+	Duration   time.Duration     `json:"duration"`
+}
+
+// HTTPRequestList represents a list of HTTP requests
+type HTTPRequestList struct {
+	Requests []HTTPRequest `json:"requests"`
+	Total    int           `json:"total"`
+	Page     int           `json:"page"`
+	PerPage  int           `json:"per_page"`
+}
```

Comment:
I know the data is still mocked at this stage, but this structure makes me think we'll end up keeping the full request and response bodies in memory for a set of requests. That might not be ideal.

Wouldn't it be better to load the bodies on demand, for example, only when the user navigates to the right panel and opens the request/response details?